src/contracts/core/AllocationManager.sol

```diff
@@ -645,6 +645,71 @@ contract AllocationManager is
         return totalMagnitude;
     }
 
+    /**
+     * @notice Returns the pending allocations of an operator for a given strategy and operatorSets
+     * @param operator the operator to get the pending allocations for
+     * @param strategy the strategy to get the pending allocations for
+     * @param operatorSets the operatorSets to get the pending allocations for
+     * @return pendingMagnitude the pending allocations for each operatorSet
+     * @return timestamps the timestamps for each pending allocation
+     */
+    function getPendingAllocations(
+        address operator,
+        IStrategy strategy,
+        OperatorSet[] calldata operatorSets
+    ) external view returns (uint64[] memory, uint64[] memory) {
+        uint64[] memory pendingMagnitude = new uint64[](operatorSets.length);
+        uint64[] memory timestamps = new uint64[](operatorSets.length);
+        for (uint256 i = 0; i < operatorSets.length; ++i) {
+            (bool exists, uint32 key, uint224 value) =
+                _magnitudeUpdate[operator][strategy][_encodeOperatorSet(operatorSets[i])].latestSnapshot();
+            if (exists) {
+                if (key > block.timestamp) {
+                    pendingMagnitude[i] = uint64(value);
+                    timestamps[i] = key;
+                } else {
```

Comment:
Would it be more useful to resize this array if you hit this else block? i.e no allocaiton for this opSet.
We can rewrite it with some assembly to resize the array.

Reply:
if I resize then how would the caller know what operatorSet it is for. basically keeping it as same size as operatorSet will help the caller figure out what opSet it is for. Unless we also return modified opSet array?